### PR TITLE
Update `winit` to `0.29`

### LINKED ISSUE #2113: `invalid setDrawableSize` on MacOS Sonoma 14.0
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

My Iced project not working on MacOS Sonoma 14.0 with the following error:

```
2023-10-12 10:29:32.396 iced-demo[3476:39547] CAMetalLayer ignoring invalid setDrawableSize width=4294967295.000000 height=4294967295.000000
thread 'main' panicked at /Users/azurice/.cargo/git/checkouts/glyphon-bce1d8fb6a1f7cb1/20f0f8f/src/text_render.rs:224:33:
attempt to add with overflow
```

And I tested some examples on branch master:
- some of them get panicked as soon as the windows appear. (tour)
- some of them are initially a blank window, and only after I resize the window it turns normal. (counter)
- Some of them seems normal. (todos)

It seems like a `winit` problem? I guess upgrading `winit` to v0.28.7 should fix it.

See https://github.com/rust-windowing/winit/issues/2876

I pasted a complete log of running example `tour` below.

### What is the expected behavior?

Panic not happens, and everything behaves normal

### Version

master

### Operating System

macOS

### Do you have any log output?

```shell
➜  iced git:(master) ✗ cargo run --package tour
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
     Running `target/debug/tour`
2023-10-12T02:59:40.223014Z  INFO wgpu_hal::gles::egl: Unable to open libEGL: Library(DlOpen { desc: "dlopen(libEGL.dylib, 0x0005): tried: \'libEGL.dylib\' (no such file), \'/System/Volumes/Preboot/Cryptexes/OSlibEGL.dylib\' (no such file), \'/usr/lib/libEGL.dylib\' (no such file, not in dyld cache), \'libEGL.dylib\' (no such file), \'/Users/azurice/Dev/iced/target/debug/build/objc_exception-eb6ed28b25f2f2fa/out/libEGL.dylib\' (no such file), \'/Users/azurice/Dev/iced/target/debug/deps/libEGL.dylib\' (no such file), \'/Users/azurice/Dev/iced/target/debug/libEGL.dylib\' (no such file), \'/Users/azurice/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libEGL.dylib\' (no such file), \'/Users/azurice/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/libEGL.dylib\' (no such file), \'/Library/Developer/CommandLineTools/usr/lib//libEGL.dylib\' (no such file)" })
2023-10-12T02:59:40.224111Z  INFO iced_wgpu::window::compositor: Settings {
    present_mode: AutoVsync,
    internal_backend: Backends(
        VULKAN | GL | METAL | DX12 | DX11 | BROWSER_WEBGPU,
    ),
    default_font: Font {
        family: SansSerif,
        weight: Normal,
        stretch: Normal,
        style: Normal,
        monospaced: false,
    },
    default_text_size: Pixels(
        16.0,
    ),
    antialiasing: None,
}
2023-10-12T02:59:40.229225Z  INFO wgpu_core::instance: Adapter Metal AdapterInfo { name: "Apple M2 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
2023-10-12T02:59:40.231728Z  INFO iced_wgpu::window::compositor: Available adapters: [
    AdapterInfo {
        name: "Apple M2 Max",
        vendor: 0,
        device: 0,
        device_type: IntegratedGpu,
        driver: "",
        driver_info: "",
        backend: Metal,
    },
]
2023-10-12T02:59:40.232673Z  INFO wgpu_core::instance: Adapter Metal AdapterInfo { name: "Apple M2 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
2023-10-12T02:59:40.232748Z  INFO iced_wgpu::window::compositor: Selected: AdapterInfo {
    name: "Apple M2 Max",
    vendor: 0,
    device: 0,
    device_type: IntegratedGpu,
    driver: "",
    driver_info: "",
    backend: Metal,
}
2023-10-12T02:59:40.233436Z  INFO iced_wgpu::window::compositor: Selected format: Bgra8UnormSrgb
2023-10-12T02:59:40.330102Z  INFO cosmic_text::font::system::std: Parsed 1431 font faces in 93ms.
2023-10-12T02:59:40.337969Z  INFO wgpu_core::device::global: Created buffer Valid((0, 1, Metal)) with BufferDescriptor { label: Some("glyphon params"), size: 16, usage: BufferUsages(COPY_DST | UNIFORM), mapped_at_creation: false }
2023-10-12T02:59:40.338663Z  INFO wgpu_core::device::global: Created texture Valid((0, 1, Metal)) with TextureDescriptor { label: Some("glyphon atlas"), size: Extent3d { width: 256, height: 256, depth_or_array_layers: 1 }, mip_level_count: 1, sample_count: 1, dimension: D2, format: Rgba8UnormSrgb, usage: TextureUsages(COPY_DST | TEXTURE_BINDING), view_formats: [] }
2023-10-12T02:59:40.339203Z  INFO wgpu_core::device::global: Created texture Valid((1, 1, Metal)) with TextureDescriptor { label: Some("glyphon atlas"), size: Extent3d { width: 256, height: 256, depth_or_array_layers: 1 }, mip_level_count: 1, sample_count: 1, dimension: D2, format: R8Unorm, usage: TextureUsages(COPY_DST | TEXTURE_BINDING), view_formats: [] }
2023-10-12T02:59:40.339835Z  INFO wgpu_core::device::global: Created buffer Valid((1, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::quad vertex buffer"), size: 32, usage: BufferUsages(VERTEX), mapped_at_creation: true }
2023-10-12T02:59:40.340357Z  INFO wgpu_core::device::global: Created buffer Valid((2, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::quad index buffer"), size: 12, usage: BufferUsages(INDEX), mapped_at_creation: true }
2023-10-12T02:59:40.345030Z  INFO wgpu_core::device::global: Created render pipeline Valid((0, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu.quad.solid.pipeline"), layout: Some((1, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 1, Metal), entry_point: "solid_vs_main" }, buffers: [VertexBufferLayout { array_stride: 8, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }] }, VertexBufferLayout { array_stride: 68, step_mode: Instance, attributes: [VertexAttribute { format: Float32x4, offset: 0, shader_location: 1 }, VertexAttribute { format: Float32x2, offset: 16, shader_location: 2 }, VertexAttribute { format: Float32x2, offset: 24, shader_location: 3 }, VertexAttribute { format: Float32x4, offset: 32, shader_location: 4 }, VertexAttribute { format: Float32x4, offset: 48, shader_location: 5 }, VertexAttribute { format: Float32, offset: 64, shader_location: 6 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 1, Metal), entry_point: "solid_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.352603Z  INFO wgpu_core::device::global: Created render pipeline Valid((1, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu.quad.gradient.pipeline"), layout: Some((2, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 2, Metal), entry_point: "gradient_vs_main" }, buffers: [VertexBufferLayout { array_stride: 8, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }] }, VertexBufferLayout { array_stride: 148, step_mode: Instance, attributes: [VertexAttribute { format: Uint32x4, offset: 0, shader_location: 1 }, VertexAttribute { format: Uint32x4, offset: 16, shader_location: 2 }, VertexAttribute { format: Uint32x4, offset: 32, shader_location: 3 }, VertexAttribute { format: Uint32x4, offset: 48, shader_location: 4 }, VertexAttribute { format: Uint32x4, offset: 64, shader_location: 5 }, VertexAttribute { format: Float32x4, offset: 80, shader_location: 6 }, VertexAttribute { format: Float32x4, offset: 96, shader_location: 7 }, VertexAttribute { format: Float32x4, offset: 112, shader_location: 8 }, VertexAttribute { format: Float32x4, offset: 128, shader_location: 9 }, VertexAttribute { format: Float32, offset: 144, shader_location: 10 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 2, Metal), entry_point: "gradient_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.353470Z  INFO wgpu_core::device::global: Created render pipeline Valid((2, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu::triangle::solid pipeline"), layout: Some((3, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 3, Metal), entry_point: "solid_vs_main" }, buffers: [VertexBufferLayout { array_stride: 24, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }, VertexAttribute { format: Float32x4, offset: 8, shader_location: 1 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 3, Metal), entry_point: "solid_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.357917Z  INFO wgpu_core::device::global: Created render pipeline Valid((3, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu.triangle.gradient.pipeline"), layout: Some((4, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 4, Metal), entry_point: "gradient_vs_main" }, buffers: [VertexBufferLayout { array_stride: 104, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }, VertexAttribute { format: Uint32x4, offset: 8, shader_location: 1 }, VertexAttribute { format: Uint32x4, offset: 24, shader_location: 2 }, VertexAttribute { format: Uint32x4, offset: 40, shader_location: 3 }, VertexAttribute { format: Uint32x4, offset: 56, shader_location: 4 }, VertexAttribute { format: Uint32x4, offset: 72, shader_location: 5 }, VertexAttribute { format: Float32x4, offset: 88, shader_location: 6 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 4, Metal), entry_point: "gradient_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.359182Z  INFO wgpu_core::device::global: Created render pipeline Valid((4, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu::image pipeline"), layout: Some((5, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 5, Metal), entry_point: "vs_main" }, buffers: [VertexBufferLayout { array_stride: 8, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }] }, VertexBufferLayout { array_stride: 36, step_mode: Instance, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 1 }, VertexAttribute { format: Float32x2, offset: 8, shader_location: 2 }, VertexAttribute { format: Float32x2, offset: 16, shader_location: 3 }, VertexAttribute { format: Float32x2, offset: 24, shader_location: 4 }, VertexAttribute { format: Sint32, offset: 32, shader_location: 5 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 5, Metal), entry_point: "fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.359235Z  INFO wgpu_core::device::global: Created buffer Valid((3, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::image vertex buffer"), size: 32, usage: BufferUsages(VERTEX), mapped_at_creation: true }
2023-10-12T02:59:40.359270Z  INFO wgpu_core::device::global: Created buffer Valid((4, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::image index buffer"), size: 12, usage: BufferUsages(INDEX), mapped_at_creation: true }
2023-10-12T02:59:40.359317Z  INFO wgpu_core::device::global: Created texture Valid((2, 1, Metal)) with TextureDescriptor { label: Some("iced_wgpu::image texture atlas"), size: Extent3d { width: 2048, height: 2048, depth_or_array_layers: 1 }, mip_level_count: 1, sample_count: 1, dimension: D2, format: Rgba8UnormSrgb, usage: TextureUsages(COPY_SRC | COPY_DST | TEXTURE_BINDING), view_formats: [] }
2023-10-12T02:59:40.372928Z  INFO wgpu_core::device::global: configuring surface with SurfaceConfiguration { usage: TextureUsages(RENDER_ATTACHMENT), format: Bgra8UnormSrgb, width: 2048, height: 1536, present_mode: AutoVsync, alpha_mode: Auto, view_formats: [] }
2023-10-12T02:59:40.372966Z  INFO wgpu_core::device::global: Automatically choosing presentation mode by rule AutoVsync. Chose Fifo
2023-10-12T02:59:40.372981Z  INFO wgpu_core::device::global: Automatically choosing alpha mode by rule Auto. Chose Opaque
2023-10-12T02:59:40.372987Z  INFO wgpu_hal::metal::surface: build swapchain SurfaceConfiguration { swap_chain_size: 3, present_mode: Fifo, composite_alpha_mode: Opaque, format: Bgra8UnormSrgb, extent: Extent3d { width: 2048, height: 1536, depth_or_array_layers: 1 }, usage: TextureUses(COLOR_TARGET), view_formats: [] }
2023-10-12T02:59:40.399901Z  INFO wgpu_core::device::global: configuring surface with SurfaceConfiguration { usage: TextureUsages(RENDER_ATTACHMENT), format: Bgra8UnormSrgb, width: 4294967295, height: 4294967295, present_mode: AutoVsync, alpha_mode: Auto, view_formats: [] }
2023-10-12T02:59:40.399943Z  WARN wgpu_core::device::global: Requested size 4294967295x4294967295 is outside of the supported range: Extent3d { width: 4, height: 4, depth_or_array_layers: 1 }..=Extent3d { width: 16384, height: 16384, depth_or_array_layers: 1 }
2023-10-12T02:59:40.399954Z  INFO wgpu_core::device::global: Automatically choosing presentation mode by rule AutoVsync. Chose Fifo
2023-10-12T02:59:40.399960Z  INFO wgpu_core::device::global: Automatically choosing alpha mode by rule Auto. Chose Opaque
2023-10-12T02:59:40.399966Z  INFO wgpu_hal::metal::surface: build swapchain SurfaceConfiguration { swap_chain_size: 3, present_mode: Fifo, composite_alpha_mode: Opaque, format: Bgra8UnormSrgb, extent: Extent3d { width: 4294967295, height: 4294967295, depth_or_array_layers: 1 }, usage: TextureUses(COLOR_TARGET), view_formats: [] }
2023-10-12 10:59:40.399 tour[4938:68323] CAMetalLayer ignoring invalid setDrawableSize width=4294967295.000000 height=4294967295.000000
2023-10-12T02:59:40.400978Z  INFO wgpu_core::device::global: Created buffer Valid((5, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::quad uniforms buffer"), size: 80, usage: BufferUsages(COPY_DST | UNIFORM), mapped_at_creation: false }
2023-10-12T02:59:40.401273Z  INFO wgpu_core::device::global: Created buffer Valid((6, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu.quad.solid.buffer"), size: 262144, usage: BufferUsages(COPY_DST | VERTEX), mapped_at_creation: false }
2023-10-12T02:59:40.401295Z  INFO wgpu_core::device::global: Created buffer Valid((7, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu.quad.gradient.buffer"), size: 524288, usage: BufferUsages(COPY_DST | VERTEX), mapped_at_creation: false }
2023-10-12T02:59:40.401393Z  INFO wgpu_core::device::global: Created buffer Valid((8, 1, Metal)) with BufferDescriptor { label: Some("glyphon vertices"), size: 4096, usage: BufferUsages(COPY_DST | VERTEX), mapped_at_creation: false }
2023-10-12T02:59:40.401416Z  INFO wgpu_core::device::global: Created buffer Valid((9, 1, Metal)) with BufferDescriptor { label: Some("glyphon indices"), size: 4096, usage: BufferUsages(COPY_DST | INDEX), mapped_at_creation: false }
2023-10-12T02:59:40.402402Z  INFO wgpu_core::device::global: Created render pipeline Valid((5, 1, Metal)) with RenderPipelineDescriptor { label: Some("glyphon pipeline"), layout: Some((0, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (0, 1, Metal), entry_point: "vs_main" }, buffers: [VertexBufferLayout { array_stride: 28, step_mode: Vertex, attributes: [VertexAttribute { format: Sint32x2, offset: 0, shader_location: 0 }, VertexAttribute { format: Uint32, offset: 8, shader_location: 1 }, VertexAttribute { format: Uint32, offset: 12, shader_location: 2 }, VertexAttribute { format: Uint32, offset: 16, shader_location: 3 }, VertexAttribute { format: Uint32, offset: 20, shader_location: 4 }, VertexAttribute { format: Float32, offset: 24, shader_location: 5 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Ccw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (0, 1, Metal), entry_point: "fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
thread 'main' panicked at /Users/azurice/.cargo/git/checkouts/glyphon-bce1d8fb6a1f7cb1/20f0f8f/src/text_render.rs:224:33:
attempt to add with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2023-10-12T02:59:40.412761Z  INFO wgpu_core::global: Dropping Global
2023-10-12T02:59:40.413049Z  INFO wgpu_core::device: Destroying 0 command encoders
```

### LINKED ISSUE #2138: Text Editor inserts the character even on the command modifier
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [x] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

The `Cmd+C` and `Cmd+V` on text editor not only copy and paste, but additionally inserts the characters `c` and `v` to the editor.

Steps to reproduce:

1. Run the `editor` example in this repository on a Macbook.
2. Select some text in the text editor.
3. Press `Cmd+C`

### What is the expected behavior?

The text editor should not be affected.

### Version

master

### Operating System

macOS

### Do you have any log output?

_No response_

### PR DESCRIPTION
**Pull request by hecrj**
This PR updates our `winit` fork with the latest changes in `0.29`.

Fixes #2113.
Fixes #2138.

### CODE REVIEW

File: winit/src/conversion.rs [hunk 1]
```diff
@@ -455,179 +449,125 @@ pub fn touch_event(
...
+            NamedKey::BrowserHome => KeyCode::WebHome,
+            NamedKey::BrowserRefresh => KeyCode::WebRefresh,
+            NamedKey::BrowserSearch => KeyCode::WebSearch,
+            NamedKey::BrowserStop => KeyCode::WebStop,
+            NamedKey::Copy => KeyCode::Copy,
+            NamedKey::Paste => KeyCode::Paste,
+            NamedKey::Cut => KeyCode::Cut,
+            _ => KeyCode::Unlabeled,
+        },
+        _ => KeyCode::Unlabeled,
```
Comment (PolyMeilex) [inline winit/src/conversion.rs:570]: Aren't we missing some of the keys? Like `Minus`, `Plus`, `Equals`.
Those are just 3 I noticed are still part of `KeyCode` enum but are never constructed.

Comment (hecrj) [reply winit/src/conversion.rs:570]: Yep, still thinking about whether we should refactor further.

**Comment by dtzxporter** [comment]
Posting here for visibility:

This also fixes the reentrant mutex issue I reported a while ago.

However, after switching to this commit, requesting redraw/redrawing an image constantly makes the window redraw really slow. It's slower the bigger the window is. The Iced debug overlay doesn't show increased times when the issue happens so it's likely something outside of iced rendering.

**Comment by dtzxporter** [comment]
I believe it's some sort of scheduling issue. There are no outliers in the flamegraph which corrosponds with iced's debug overlay not having increased times. Even the spinner example gets choppy sometimes now on the winit branch and it doesn't on the latest master.

**Comment by planetoryd** [comment]
Yeah please do this gets the window shadow drawn

**Comment by planetoryd** [comment]
No, in `cargo tree` it shows  sctk-adwaita is at `0.7.0`.

 sctk-adwaita at v0.8.0 has shadows.

**Comment by PolyMeilex** [comment]
> No, in `cargo tree` it shows sctk-adwaita is at `0.7.0`.
> 
> sctk-adwaita at v0.8.0 has shadows.

`winit 0.29.4` depends on `sctk-adwaita 0.8`, so not sure what has happend in your case, perhaps try `cargo update` to bump your cargo lock

**Comment by PolyMeilex** [comment]
Nevermind that, I forgot that iced is using a fork of winit (that does not use 0.8 yet): https://github.com/iced-rs/winit/blob/3bcdb9abcd7459978ec689523bc21943d38da0f9/Cargo.toml#L163

**Comment by elevenhsoft** [comment]
Updating `winit` to work with `sctk-adwaita` > `0.8.0` require adding only 1 arguments like so

image

**Pull request merged**